### PR TITLE
Fix errors reported by Coverity

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?
Just fix errors reported by Coverity.
Follow up #1019 

## How was this patch tested?

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by LuciferYang** [comment]
@zhixingheyi-tian I tend to split this PR into two patches， one about oap code, the other abouts spark source code

File: src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
   /**
    * Repetition/Definition/Value readers.
    */
-  private SpecificParquetRecordReaderBase.IntIterator repetitionLevelColumn;
+//  private SpecificParquetRecordReaderBase.IntIterator repetitionLevelColumn;
```
Comment (LuciferYang) [inline src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java]: I tend to keep this code part, `SkippableVectorizedColumnReader` already override `readPageV1` and `readPageV2` method, and not init this field, I suggest push this change to spark first

File: src/main/java/org/apache/parquet/hadoop/ParquetFiberDataReader.java [hunk 2]
```diff
@@ -288,7 +289,8 @@ private PageHeader readPageHeader() throws IOException {
           default:
             LOG.debug("skipping page of type {} of size {}",
               pageHeader.getType(), compressedPageSize);
-            this.skip(compressedPageSize);
+            long k = this.skip(compressedPageSize);
+            if(k <= 0) LOG.warn("invalid skip,k =" + k);
```
Comment (LuciferYang) [inline src/main/java/org/apache/parquet/hadoop/ParquetFiberDataReader.java]: maybe `LOG.warn("invalid skip,k {}", k)` better than `LOG.warn("invalid skip,k =" + k)`

File: src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java [hunk 3]
```diff
@@ -165,7 +165,7 @@ public void readBatch(int total, WritableColumnVector column) throws IOException
         leftInPage = (int) (endOfPageValueCount - valuesRead);
       }
       int num = Math.min(total, leftInPage);
-      if (isCurrentPageDictionaryEncoded) {
+      if (isCurrentPageDictionaryEncoded && (dictionaryIds != null)) {
```
Comment (LuciferYang) [inline src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java]: Need a test case to produce `dictionaryIds  == null`  && `dictionary != null` && `isCurrentPageDictionaryEncoded is true`, want to know when this scenario will appear

File: src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java [hunk 4]
```diff
@@ -532,6 +532,7 @@ private void readFixedLenByteArrayBatch(
 
   protected void readPage() {
     DataPage page = pageReader.readPage();
+    if(null == page) return;
```
Comment (LuciferYang) [inline src/main/spark2.3.2/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java]: if (page == null) is true I think we should throw a IOE instead of silent return, `readPage` not a public method, if there is not page, the invocation logic is wrong already

File: src/main/java/org/apache/parquet/hadoop/MrOapRecordReader.java [hunk 5]
```diff
@@ -62,11 +62,17 @@ public T getCurrentValue() throws IOException, InterruptedException {
...
-      this.internalReader = new InternalParquetRecordReader<>(readSupport);
-      this.internalReader.initialize(parquetFileReader, configuration);
+      ParquetFileReader parquetFileReader = null;
+      try {
+        parquetFileReader =
+                ParquetFileReader.open(configuration, file, footer.toParquetMetadata());
+        parquetFileReader.filterRowGroups(getFilter(configuration));
+        this.internalReader = new InternalParquetRecordReader<>(readSupport);
+        this.internalReader.initialize(parquetFileReader, configuration);
+      }catch (IOException e) {
```
Comment (LuciferYang) [inline src/main/java/org/apache/parquet/hadoop/MrOapRecordReader.java]: `}catch` => `} catch`

**Comment by zhixingheyi-tian** [comment]
@LuciferYang 

Thanks your review.
